segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DurableLog.java

```diff
@@ -309,7 +303,8 @@ public int getId() {
 
     //region Recovery
 
-    private boolean performRecovery() throws Exception {
+    @SneakyThrows
```

Comment:
Hmm, interesting. It looks like both the contents of the `try` and `finally` blocks can throw. I have rewritten the code so it wraps both and rethrows as a `CompletionException`